src/charm.py

```diff
@@ -90,150 +139,158 @@ def __init__(self, *args):
             self.on[FIVEG_N4_RELATION_NAME].relation_broken,
             self._configure_sdcore_nms,
         )
+        # Handling config changed event to publish the new url if the unit reboots and gets new IP
+        self.framework.observe(self.on.config_changed, self._configure_sdcore_nms)
+        self._webui = Webui(url="")
 
     def _configure_sdcore_nms(self, event: EventBase) -> None:
-        """Add Pebble layer and manages Juju unit status.
+        """Handle Juju events.
 
-        Args:
-            event (EventBase): Juju event.
+        Whenever a Juju event is emitted, this method performs a couple of checks to make sure that
+        the workload is ready to be started. Then, it configures the NMS workload,
+        runs the Pebble services and expose the service information through charm's interface.
         """
         if not self._container.can_connect():
             return
-        if not self._container.exists(path=CONFIG_DIR_PATH):
+        if not self._container.exists(path=BASE_CONFIG_PATH):
             return
-        self._configure_upf_information()
-        self._configure_gnb_information()
-        if not self.model.relations.get(SDCORE_MANAGEMENT_RELATION_NAME):
+        for relation in [COMMON_DATABASE_RELATION_NAME, AUTH_DATABASE_RELATION_NAME]:
+            if not self._relation_created(relation):
+                return
+        if not self._common_database_resource_is_available():
             return
-        if not self._sdcore_management.management_url:
+        if not self._auth_database_resource_is_available():
             return
-        self._configure_pebble()
+        self._setup_webui_endpoint_url()
+        self._configure_gnbs()
+        self._configure_upfs()
+        desired_config_file = self._generate_webui_config_file()
+
+        if config_update_required := self._is_config_update_required(desired_config_file):
+            self._write_file_in_workload(WEBUI_CONFIG_PATH, desired_config_file)
```

Comment:
We could move those 3 in a function to simplify the function here, we'd have something like

```python
if not x:
    return
if not y:
    return
do_a()
do_b()
do_c()
```

Reply:
I took this from the webui and it's the same behavior that we have in all the charms. 
I was able to simplify it and get rid of the `restart` boolean.
However, I inverted the order, now we create the pebble plan and then write the configuration file. I think it does not break anything.

Reply:
not true, we need the file to be written before creating the pebble layer. need to fix this

Reply:
done